spec/models/user_spec.rb

```diff
@@ -2046,6 +2046,10 @@ def hash(password, salt)
     fab!(:category3) { Fabricate(:category) }
     fab!(:category4) { Fabricate(:category) }
 
+    # UGLY but perf is horrible with this callback
+    before { User.set_callback(:create, :after, :ensure_in_trust_level_group) }
+    after { User.skip_callback(:create, :after, :ensure_in_trust_level_group) }
```

Comment:
Is this related to the change here? If not, I think splitting out into another PR where the before/after timings are provided is more useful.

Reply:
Yes, else the tests will fail since the `ensure_in_trust_level_group` callback is skipped by default.

Reply:
Can we perhaps run `Group.user_trust_level_change!(id, trust_level)` for the user manually and only for the test which you've added?

Reply:
It's impossible since I need to run this before the `after_create` callback `set_default_categories_preferences`. Instead, I can create the record directly using the `CategoryUser.create` method.